packages/app/src/components/commons/WalletBadge.tsx

```diff
@@ -12,8 +20,34 @@ const WalletAddressContainer = styled(Box)({
   boxSizing: "border-box",
 })
 
-export const WalletBadge: React.FC<{ address: string; hover?: boolean }> = ({ address, hover }) => {
+export const WalletBadge: React.FC<WalletBadgeProps> = ({ address, hover, copyable }) => {
   const avatarSrc = blockies.create({ seed: address.toLowerCase() }).toDataURL()
+  const provider = new ethers.providers.Web3Provider(window.ethereum)
```

Comment:
This is unsafe since we don't know if its a browser wallet (like MetaMask). If not `windown.ethereum` might now be available. We need to get the provider that we are already using. 
Something like this:
```suggestion
    const { connector, active } = useWeb3React()
    // check if active, then
    const provider = await connector?.getProvider()
    const web3Provider = new ethers.providers.Web3Provider(provider)
    const ensName = await web3Provider.lookupAddress("0x..")
```

Reply:
Done